src/lib/Scenes/Search2/Search2.tsx

```diff
@@ -1,10 +1,168 @@
-import { Flex, Text } from "palette"
-import React from "react"
+import { captureMessage } from "@sentry/react-native"
+import { Search2Query, Search2QueryResponse } from "__generated__/Search2Query.graphql"
+import { AboveTheFoldFlatList } from "lib/Components/AboveTheFoldFlatList"
+import { ArtsyKeyboardAvoidingView } from "lib/Components/ArtsyKeyboardAvoidingView"
+import OpaqueImageView from "lib/Components/OpaqueImageView/OpaqueImageView"
+import { SearchInput } from "lib/Components/SearchInput"
+import { navigate } from "lib/navigation/navigate"
+import { defaultEnvironment } from "lib/relay/createEnvironment"
+import { isPad } from "lib/utils/hardware"
+import { useAlgoliaClient } from "lib/utils/useAlgoliaClient"
+import { Flex, Sans, Spacer, Touchable, useColor } from "palette"
+import React, { useRef, useState } from "react"
+import { connectHighlight, connectInfiniteHits, connectSearchBox, InstantSearch } from "react-instantsearch-native"
+import { FlatList, Platform, ScrollView } from "react-native"
+import { graphql, QueryRenderer } from "react-relay"
+import styled from "styled-components"
+import { CityGuideCTA } from "../Search/CityGuideCTA"
+
+interface ImprovedSearchInputProps {
+  refine: (value: string) => any
+}
+
+const ImprovedSearchInput: React.FC<ImprovedSearchInputProps> = ({ refine }) => {
+  return (
+    <SearchInput
+      placeholder="Search artists"
+      onChangeText={(queryText) => {
+        refine(queryText)
+      }}
+    />
+  )
+}
+
+const Highlight = connectHighlight(({ highlight, attribute, hit, highlightProperty = "_highlightResult" }) => {
+  const parsedHit = highlight({ attribute, hit, highlightProperty })
 
-export const Search2: React.FC = () => {
   return (
-    <Flex flexGrow={1} justifyContent="center" alignItems="center">
-      <Text>Improved Search Screen</Text>
-    </Flex>
+    <Sans size="3t" weight="regular">
+      {parsedHit.map(({ isHighlighted, value }) =>
+        isHighlighted ? (
+          <Sans size="3t" weight="medium" style={{ padding: 0, margin: 0 }}>
```

Comment:
fyi: I believe these components can take `padding` and `margin` directly though I'm a bit surprised that they would have initial non-zero values.

```suggestion
          <Sans size="3t" weight="medium" padding={0} margin={0} >
```